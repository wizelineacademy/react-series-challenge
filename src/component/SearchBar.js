import React, {Component} from 'react'
import styled from 'styled-components'

const Div = styled.div`
  text-align: center;
`

const Input = styled.input`
  padding: 0.5em;
  margin: 0.5em;
  border: 1px solid #ccc;
  color: gray
  border-radius: 3px;
  width: 50%;
`

class SearchBar extends Component {
  constructor (props) {
    super(props)

    this.state = {
      searchValue: ''
    }
  }

  handleOnchange(event) {
    this.setState(
      {searchValue: event.target.value},
      () => {this.props.searchFunction(this.state.searchValue)}
    )
  }

  render () {
    return (
      <Div>
        <Input
          className='input'
          type='text'
          value={this.state.searchValue}
          onChange={(event) => this.handleOnchange(event)}
          placeholder={this.props.placeholder}
        />
      </Div>
    )
  }
}

export default SearchBar
