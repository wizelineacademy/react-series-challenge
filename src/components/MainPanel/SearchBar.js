import React, { Component } from 'react';
import { bindActionCreators } from "redux";
import actions from "../../actions";
import { connect } from "react-redux";

import StyledSearchBar from '../styled/SearchBar.styled';

const { searchGifs } = actions.creators;

class SearchBar extends Component {

  componentDidMount() {
    this.input.value = this.props.searchQuery;
  }

  render () {
    return (
      <StyledSearchBar 
        onChange={(e) => this.props.searchGifs(e.target.value)} 
        type='text' 
        placeholder='Search'
        ref={el => this.input = el}
      />
    )
  }
}

const mapStateToProps = state => ({
  searchQuery: state.searchQuery
})

const mapDispatchToProps = dispatch => (
  bindActionCreators({ searchGifs }, dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(SearchBar);