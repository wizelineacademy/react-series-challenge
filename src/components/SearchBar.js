import React, { Component } from 'react';
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux';

import { SearchBarInput } from '../styles/SearchBar.style.js';

import gifActions from '../actions/gifActions';

export class SearchBar extends Component {
  render() {
    const { searchGifs } = this.props;
    const self = this;

    return (
      <div>
        <SearchBarInput
          type="text"
          placeholder="Search gifs!!"
          innerRef={ node => { this.input = node } }
          onKeyPress={ event => {
            if(event.key === 'Enter' && self.input.value.trim()) {
              searchGifs(self.input.value);
            }
          }}
        />
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => {
  const { searchGifs } = gifActions.creators;

  return bindActionCreators({ searchGifs }, dispatch);
}

export default connect(null, mapDispatchToProps)(SearchBar);
