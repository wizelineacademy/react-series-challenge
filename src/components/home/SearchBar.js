/**
 * SearchBar Component
 */

import React, { Component } from 'react';

/**
 * SearchBar Component becomes a class based component to handle its value state
 */

export default class SearchBar extends Component {
    constructor(props){
        super(props);

        this.state = {
            term: ""
        }
    }

    onTermChange = (e) => {
        const targetElement = e.target;
        const newValue = targetElement.value;
        
        this.setState( () => {
            return { term: newValue }
        } );
    }

    render(){
        return (
            <section className="SearchBar">
                <input type="text" value={this.state.term} onChange={ this.onTermChange } placeholder="Search Giphy:" />
            </section>
        );
    };
}