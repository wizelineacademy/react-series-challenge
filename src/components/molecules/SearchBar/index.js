import React from "react";
import styled from "styled-components";
import Input from "../../atoms/Input/index";

const StyledSearchBar = styled.div`
  margin: 0 auto;
  max-width: 700px;
  display: table;
`;

const SearchBar = props => {
  return (
    <StyledSearchBar>
      <Input
        getInputValue={props.getInputValue}
        placeholder="Type some words..."
      />
    </StyledSearchBar>
  );
};

export default SearchBar;
