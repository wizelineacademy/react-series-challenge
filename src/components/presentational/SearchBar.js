import React from 'react';
import styled from 'styled-components';

const SearchInput = styled.input`
  width: 100%;
  padding: 5px;
  margin-bottom: 15px;
  font-size: 1.2em;
  border: none;
  border-bottom: 1px solid #ccc;
`;

const SearchBar = (props) => {
  return (
    <SearchInput
      type="search"
      placeholder="Start to type for searching..."
      onChange={props.searchFunction}
     />
  );
}

export default SearchBar;