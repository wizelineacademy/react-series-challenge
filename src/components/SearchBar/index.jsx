import React from 'react'
import { func, string } from 'prop-types'
import Icon from 'react-icons-kit'
import { search } from 'react-icons-kit/icomoon/search'

import { StyledInput, StyledForm } from './SearchBar.style'

const SearchBar = ({ onChange, onSubmit, value }) => (
  <div>
    <StyledForm onSubmit={onSubmit}>
      <StyledInput onChange={onChange} value={value} type='text' name='search' />
      <button type='submit' ><Icon icon={search} /></button>
    </StyledForm>
  </div>
)

SearchBar.propTypes = {
  onChange: func.isRequired,
  onSubmit: func.isRequired,
  value: string
}
SearchBar.defaultProps = {
  value: ''
}
export default SearchBar
