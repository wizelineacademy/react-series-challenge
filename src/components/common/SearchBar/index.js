import SearchBar from './StyledSearchBar';

export default SearchBar
