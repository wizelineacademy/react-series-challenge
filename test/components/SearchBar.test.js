import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import SearchBar from '../../src/components/SearchBar';

const ENTER_KEY = {
  keyCode: 13,
  which: 13,
  key: 'enter'
};
const NOT_ENTER_KEY = {
  keyCode: 14,
  which: 14,
  key: 'not_enter'
};

describe('SearchBar', () => {
  let props;
  beforeEach(() => {
    props = {
      handleSubmit: jest.fn(),
      history: [],
    }
  })

  it('should render correctly', () => {
    const tree = renderer
    .create(
      <SearchBar {...props} />
    ).toJSON();

    expect(tree).toMatchSnapshot();  
  });

  it('should call the handleSubmit callback when enter key is pressed', () => {
    document.body.innerHTML = '<input id="search-bar" value="" />';
    const output = shallow(
      <SearchBar {...props} />, { attachTo: document.body }
    );
    expect(props.handleSubmit.mock.calls.length).toBe(0);
    output.find('#search-bar').simulate('keyUp', NOT_ENTER_KEY);
    output.find('#search-bar').simulate('keyUp', ENTER_KEY);
    expect(props.handleSubmit.mock.calls.length).toBe(1);
  });
  
});