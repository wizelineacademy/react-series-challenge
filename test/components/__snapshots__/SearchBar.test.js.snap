// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 70px;
}

.c1 {
  position: absolute;
}

.c1 input {
  border: none;
  background: none;
  z-index: 1;
  width: 50px;
  height: 50px;
  -webkit-transition: all .25s ease-in .25s;
  transition: all .25s ease-in .25s;
  color: transparent;
  font-size: 1.25rem;
  line-height: 50px;
}

.c1 input:hover {
  cursor: pointer;
}

.c1 input:hover:focus {
  cursor: text;
}

.c1 input:hover + span {
  background: rgba(255,255,255,0.2);
}

.c1 input:focus {
  width: 200px;
  padding: 0 10px;
  outline: none;
  color: black;
  background: none;
  color: white;
}

.c1 input:focus + span {
  width: 200px;
}

.c1 input:focus + span::before {
  width: 2px;
  opacity: 0;
  -webkit-transition: all .25s ease-in;
  transition: all .25s ease-in;
}

.c1 input + span {
  z-index: -1;
  position: absolute;
  border: 2px solid white;
  top: 0;
  width: 50px;
  height: 50px;
  -webkit-transition: all .25s ease-in .25s;
  transition: all .25s ease-in .25s;
  border-radius: 50px;
  left: 0;
}

.c1 input + span::before {
  -webkit-transition: all .25s ease-in .5s;
  transition: all .25s ease-in .5s;
  -webkit-transform-origin: left top;
  -ms-transform-origin: left top;
  transform-origin: left top;
  content: '';
  position: absolute;
  width: 20px;
  height: 7px;
  border-radius: 5px;
  background: white;
  -webkit-transform: rotate(45deg) translate(43px,10px);
  -ms-transform: rotate(45deg) translate(43px,10px);
  transform: rotate(45deg) translate(43px,10px);
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <input
      id="search-bar"
      onKeyUp={[Function]}
      type="search"
    />
    <span />
  </div>
</div>
`;
